XLS-30/README.md

```diff
@@ -0,0 +1,1255 @@
+     - Narrows time windows in which the pool suffers decreased trading volume
+     - Liquidity providers additionally reap a share of profits that would otherwise go JUST to arbitrageurs
+
+## 2. Creating AMM instance on XRPL
+
+### 2.1. On-Ledger Objects
+
+We propose the existing **`AccountRoot`** object together with a new ledger object **`AMM`** to represent an AMM instance. Currently, the **`AccountRoot`** object type describes a single account, its settings, and XRP balance on the XRP ledger. The **`AccountRoot`** and the **`AMM`** objects that represent an AMM instance can be created using the new **`AMMCreate`** transaction. XRP balance of the AMM instance pools is always tracked via the existing `Balance` field of the **`AccountRoot`** object. The issued asset balances of the AMM instance pools and **`LPTokens`** are automatically tracked via trustlines. The AMM can be traded against using the new **`AMMDeposit`** and **`AMMWithdraw`** and the existing **`Payment`** transactions. Additionally, the AMM instance in "empty" state can be deleted using the `AMMDelete` transaction.
+
+Note that in this version, only equal-weighted two asset pools are supported. However, differing weighted pools could be supported in future versions.
+
+#### 2.1.1. Ledger Entries representing AMM instance
+
+##### **`AccountRoot`** ledger entry
+
+A new flag **`lsfAMM`** is introduced to the [**`AccountRoot`**](https://xrpl.org/accountroot.html) object. 
+
+| Flag Name |  Hex Value | Description |
+| ---------- | :----------------: | :-------:|
+| `lsfAMM`      | 0x40000000  |   If set, it indicates that the object represents an AMM instance|
+
+Additionally, we introduce a new field called `AMMID` to the `AccountRoot` object. It replaces directory entry to link account root and AMM object. 
+
+| Field Name        |     Required?      | JSON Type | Internal Type |
+| ----------------- | :----------------: | :-------: | :-----------: |
+| `AMMID` | :heavy_check_mark: | `string`  |   `UINT256`    |
+
+`AMMID` specifies the `AMMID` as described below.
+
+The unique ID of the **`AccountRoot`** object, a.k.a. **`AccountRootID`** is computed as follows:
+- for (i = 0; i <= 256; i--)
+  - Compute `AccountRootID` = `SHA512-Half`(i || [Parent Ledger Hash](https://xrpl.org/ledgerhashes.html) || `AMMID`)
+  - If the computed `AccountRootID` exists, repeat
+    - else, return `AccountRootID`
+ 
+#### **`AMM`** ledger entry 
+The unique ID of the new **`AMM`** object , a.k.a **`AMMID`** is computed as follows:
+- Calculate the `SHA512-Half` of some of the following values:
+  - The `issuer` of the issued asset;
+  - The `currency` of the issued asset;
+  - The `issuer` of the second issued asset, if there exists one;
+  - The `currency` of the second issued asset, if there exists one;
+  <!-- `XRP` if the `Balance` field exists, i.e. if one of the assets is XRP "GT: why is this needed? We just use issuer/currency of each asset. XRP defaults both to 0."-->
```

Comment:
The serialization format is documented here: https://xrpl.org/serialization.html#currency-codes

> The currency code `0x0000000000000000000000000000000000000000` (all zeroes) is generally disallowed. Usually, XRP amounts are not specified with currency codes. However, this code is used to indicate XRP in rare cases where a field must specify a currency code for XRP.